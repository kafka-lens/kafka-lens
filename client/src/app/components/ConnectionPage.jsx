import React from 'react';
import '../css/ConnectionPage.scss';
import lensIcon from '../../../../assets/images/lens-icon.png';

const ConnectionPage = ({
  isFetching, connected, updateURI, validConnectionChecker,
}) => {
  let loading;
  if (isFetching === true) {
    loading = (
      <div className="loading">
        <div className="loading-bar" />
        <div className="loading-bar" />
        <div className="loading-bar" />
        <div className="loading-bar" />
      </div>
    );
  }

  // This handles connection timeouts for Kafka cluster
  let errorMsg;
  if (connected === false) {
    errorMsg = <p id="error-message">Connection timed out, please check your connection URI.</p>;
  }
  return (
    <div className="connection-box">
      <div className="connection-form container">
        <form>
          <div>
            <p id="label-instructions">Enter Your Kafka Server URI</p>
            <input id="uri-input" type="text" onChange={updateURI} />
          </div>
          <button
            id="connect-button"
            className="btn waves-effect waves-light"
            type="submit"
            onClick={validConnectionChecker}
          >
            Connect
          </button>
        </form>
        {/* Error message displays here */}
        {errorMsg}
      </div>
      {/* Loading bars here */}
      <div className="loading-bars">{loading}</div>
      <footer className="footer">
        <img src={lensIcon} alt="" />
        <p className="FooterText">Â© Kafka Lens Version 2.0 </p>
      </footer>
    </div>
  );
};

export default ConnectionPage;
