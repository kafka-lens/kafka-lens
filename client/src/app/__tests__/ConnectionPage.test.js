import React from 'react';
import { shallow, mount } from 'enzyme';

import ConnectionPage from '../components/ConnectionPage';

describe('Connection.js unit tests', () => {
  let wrapper;
  const props = {
    connected: null,
    isFetching: false,
    updateURI: jest.fn(),
    validConnectionChecker: jest.fn(),
  };
  beforeAll(() => {
    wrapper = shallow(<ConnectionPage {...props} />);
  });
  it('Renders without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should contain a Connect button', () => {
    expect(
      wrapper.containsMatchingElement(
        <button
          id="connect-button"
          className="btn waves-effect waves-light"
          type="submit"
          onClick={props.validConnectionChecker}
        >
          Connect
        </button>,
      ),
    ).toBeTruthy();
  });

  it('Should invoke validConnectionCheck on click', () => {
    wrapper
      .find('button')
      .at(0)
      .simulate('click');
    expect(props.validConnectionChecker).toHaveBeenCalled();
  });

  it('Should invoke updateURI on change', () => {
    wrapper
      .find('input')
      .at(0)
      .simulate('change');
    expect(props.updateURI).toHaveBeenCalled();
  });

  it('Should display error message if props.connected is false', () => {
    wrapper.setProps({ connected: false });
    expect(
      wrapper.containsMatchingElement(
        <p id="error-message">Connection timed out, please check your connection URI.</p>,
      ),
    ).toBeTruthy();
  });

  it('Should not display error message if props.connected is not false', () => {
    wrapper.setProps({ connected: null });
    expect(
      wrapper.containsMatchingElement(
        <p id="error-message">Connection timed out, please check your connection URI.</p>,
      ),
    ).toBeFalsy();
  });

  it('Should not display loading bar if props.isFetching is false', () => {
    expect(
      wrapper.containsMatchingElement(
        <div className="loading">
          <div className="loading-bar" />
          <div className="loading-bar" />
          <div className="loading-bar" />
          <div className="loading-bar" />
        </div>,
      ),
    ).toBeFalsy();
  });

  it('Should display loading bar if props.isFetching is true', () => {
    wrapper.setProps({ isFetching: true });
    expect(
      wrapper.containsMatchingElement(
        <div className="loading">
          <div className="loading-bar" />
          <div className="loading-bar" />
          <div className="loading-bar" />
          <div className="loading-bar" />
        </div>,
      ),
    ).toBeTruthy();
  });
});
