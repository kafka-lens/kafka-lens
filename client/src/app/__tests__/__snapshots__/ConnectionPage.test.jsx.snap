// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection.js unit tests Renders without errors 1`] = `
<div
  className="connection-box"
>
  <div
    className="connection-form container"
  >
    <form>
      <div>
        <p
          id="label-instructions"
        >
          Enter Your Kafka Server URI
        </p>
        <input
          id="uri-input"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
      <button
        className="btn waves-effect waves-light"
        id="connect-button"
        onClick={[MockFunction]}
        type="submit"
      >
        Connect
      </button>
    </form>
  </div>
  <div
    className="loading-bars"
  />
  <footer
    className="footer"
  >
    <img
      alt=""
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <p
      className="FooterText"
    >
      Â© Kafka Lens Version 2.0 
    </p>
  </footer>
</div>
`;
